qiskit_experiments/framework/base_experiment.py

```diff
@@ -167,29 +144,114 @@ def _initialize_experiment_data(
 
         return experiment_data._copy_metadata()
 
-    def run_analysis(self, experiment_data, **options) -> ExperimentData:
+    def _pre_transpile_hook(self, backend: Backend):
+        """An extra subroutine executed before transpilation.
+
+        Args:
+            backend: Target backend.
+        """
+        pass
+
+    # pylint: disable = unused-argument
+    def _post_transpile_hook(
+        self, circuits: List[QuantumCircuit], backend: Backend
+    ) -> List[QuantumCircuit]:
+        """An extra subroutine executed after transpilation.
```

Comment:
Can you mention the subclass may chose to implement this and give an example of what could go in this? Is it assumed that any needed arguments would either come from the backend or the options?

Reply:
Good suggestion. Done in https://github.com/Qiskit/qiskit-experiments/pull/380/commits/88f1304f7e8fead332f21204551ec056aa8c8ba2